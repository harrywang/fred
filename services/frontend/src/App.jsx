import React, { useState, useEffect } from 'react'
import axios from 'axios'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import NavBar from './components/NavBar'
import Home from './components/Home'
import Login from './components/Login'
import Register from './components/Register'
import Status from './components/Status'
import Dashborad from './components/Dashboard'
import UserStatus from './components/UserStatus'
import { Layout, message } from 'antd'
import styles from './App.module.scss'
import Image404 from './img/404.svg'
import './App.scss'

const App = () => {
  const [users, setUsers] = useState([])
  const [accessToken, setAccessToken] = useState(null)

  const getUsers = () => {
    axios
      .get(`${process.env.REACT_APP_BACKEND_SERVICE_URL}/users`)
      .then(res => {
        setUsers(res.data)
      })
      .catch(err => {
        console.log(err)
      })
  }

  const validRefresh = () => {
    const token = window.localStorage.getItem('refreshToken')
    if (token) {
      axios
        .post(`${process.env.REACT_APP_BACKEND_SERVICE_URL}/auth/refresh`, {
          refresh_token: token,
        })
        .then(res => {
          setAccessToken(res.data.access_token)
          getUsers()
          window.localStorage.setItem('refreshToken', res.data.refresh_token)
          return true
        })
        .catch(err => {
          console.log(err)
          return false
        })
    }
    return false
  }

  const isAuthenticated = () => {
    if (accessToken || validRefresh()) {
      return true
    }
    return false
  }

  const handleRegisterFormSubmit = async data => {
    const url = `${process.env.REACT_APP_BACKEND_SERVICE_URL}/auth/register`
    return axios
      .post(url, data)
      .then(res => {
        return res
      })
      .catch(err => {
        return err
      })
  }

  const handleLoginFormSubmit = async data => {
    const url = `${process.env.REACT_APP_BACKEND_SERVICE_URL}/auth/login`
    return axios
      .post(url, data)
      .then(res => {
        setAccessToken(res.data.access_token)
        getUsers()
        window.localStorage.setItem('refreshToken', res.data.refresh_token)
        return res
      })
      .catch(err => {
        return err
      })
  }

  const logoutUser = () => {
    window.localStorage.removeItem('refreshToken')
    setAccessToken(null)
    message.success('You have logged out.')
  }

  const PageNotFound = () => (
    <div className={styles.not_found}>
      <div className={styles.not_found_title}>Oops, Page Not Found!</div>

      <div className={styles.not_found_image}>
        <img src={Image404} width="50%" alt="404 Page Not Found" />
      </div>
    </div>
  )

  const { Header, Content, Footer } = Layout

  return (
    <Router>
      <Layout style={{ minHeight: '100vh' }}>
        <Header style={{ background: '#ffffff', display: 'flex' }}>
          <NavBar isAuthenticated={isAuthenticated} logoutUser={logoutUser} />
        </Header>

        <Content>
          <Switch>
            <Route exact path="/">
              <Home />
            </Route>
            <Route exact path="/login">
              <Login
                isAuthenticated={isAuthenticated}
                handleLoginFormSubmit={handleLoginFormSubmit}
              />
            </Route>
            <Route exact path="/register">
              <Register handleRegisterFormSubmit={handleRegisterFormSubmit} />
            </Route>
            <Route exact path="/status">
              <Status
                isAuthenticated={isAuthenticated}
                accessToken={accessToken}
              />
            </Route>
            <Route exact path="/dashboard">
              <Dashborad isAuthenticated={isAuthenticated} />
            </Route>
            <Route exact path="/status">
              <UserStatus />
            </Route>
            <Route component={PageNotFound} />
          </Switch>
        </Content>

        <Footer
          style={{
            textAlign: 'center',
            background: '#444F60',
            padding: '3rem 1.5rem 6rem',
            color: '#ffffff',
          }}
        >
          Made with <i className="fa fa-heart pulse" /> by{' '}
          <a
            href="http://harrywang.me"
            target="_blank"
            rel="noopener noreferrer"
          >
            Harry Wang
          </a>
        </Footer>
      </Layout>
    </Router>
  )
}

export default App
