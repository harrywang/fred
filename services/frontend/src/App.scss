html,
body {
  font-family: 'Open Sans', sans-serif;
  font-size: 14px;
}
