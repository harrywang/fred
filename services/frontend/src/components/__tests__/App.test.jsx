import React from 'react'
import { cleanup } from '@testing-library/react'

import App from '../../App'

afterEach(cleanup)

test('renders', () => {
  const { asFragment } = renderWithRouter(<App />)
  expect(asFragment()).toMatchSnapshot()
})
