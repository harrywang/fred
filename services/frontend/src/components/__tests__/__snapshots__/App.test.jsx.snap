// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <section
    class="ant-layout"
    style="min-height: 100vh;"
  >
    <header
      class="ant-layout-header"
      style="background: rgb(255, 255, 255); display: flex;"
    >
      <div
        class="navbar_wrap"
      >
        <div
          class="navbar_logo"
        >
          Fred
        </div>
        <ul
          class="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
          role="menu"
        >
          <li
            class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
            role="menuitem"
            style="display: none;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
            >
              <span>
                ···
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
          <li
            class="ant-menu-item ant-menu-item-only-child"
            role="menuitem"
          >
            <a
              data-testid="nav-register"
              href="/register"
            >
              Register
            </a>
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
            role="menuitem"
            style="display: none;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
            >
              <span>
                ···
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
          <li
            class="ant-menu-item ant-menu-item-only-child"
            role="menuitem"
          >
            <a
              data-testid="nav-login"
              href="/login"
            >
              Log In
            </a>
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
            role="menuitem"
            style="visibility: hidden; position: absolute; display: none;"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
            >
              <span>
                ···
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
          </li>
        </ul>
      </div>
    </header>
    <main
      class="ant-layout-content"
    >
      <div>
        <div
          class="home_welcome_wrap"
        >
          <div
            class="home_welcome"
          >
            <div
              class="ant-row ant-row-center"
            >
              <div
                class="ant-col ant-col-12"
              >
                <div
                  class="home_welcome_text"
                >
                  <div
                    class="home_welcome_text_title"
                  >
                    Meet FRED: Flask + REact + Docker
                  </div>
                  <div
                    class="home_welcome_text_desc"
                  >
                    An End-to-End Boilerplate for Full Stack Development
                  </div>
                  <div
                    class="home_welcome_text_button_wrap"
                  >
                    <button
                      class="ant-btn ant-btn-primary ant-btn-round ant-btn-lg"
                      style="margin-right: 8px;"
                      type="button"
                    >
                      <span>
                        What Fred uses?
                      </span>
                    </button>
                    <button
                      class="ant-btn ant-btn-round ant-btn-lg"
                      type="button"
                    >
                      <span>
                        Where is Fred?
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-12"
              >
                <div
                  class="home_welcome_img"
                >
                  <a
                    href="https://undraw.co/"
                  >
                    <img
                      alt="fred at work"
                      src="/img/illustrations/designer.svg"
                    />
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="use_wrap"
        >
          <div
            class="use_title_wrap"
          >
            <div
              class="use_title"
            >
              What Fred Uses
            </div>
            <div
              class="use_subtitle"
            >
              to make you happy
            </div>
            <div
              class="use_divider"
            />
          </div>
          <div
            class="icon_wrap"
          >
            <div
              class="ant-row ant-row-space-around"
              style="margin: -8px -8px 8px -8px;"
            >
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="flask-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Flask
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="logo-react"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    React
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="cube-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Docker
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="server-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Postgres
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row ant-row-space-around"
              style="margin: -8px -8px 8px -8px;"
            >
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="layers-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    SQLAlchemy
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="logo-css3"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Bulma
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="git-compare-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    CircleCI
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="rocket-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Heroku
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row ant-row-space-around"
              style="margin: -8px -8px 8px -8px;"
            >
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="logo-nodejs"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Nodejs
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="logo-npm"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    NPM
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="flashlight-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Jest
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="reader-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Formik
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row ant-row-space-around"
              style="margin: -8px -8px 8px -8px;"
            >
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="logo-python"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Python
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="hammer-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    PyTest
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="key-outline"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    JWT
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-3"
                style="padding: 8px 8px 8px 8px;"
              >
                <div
                  class="icon_item_wrap"
                >
                  <div
                    class="icon_item"
                  >
                    <span
                      class="anticon"
                      role="img"
                    >
                      <ion-icon
                        name="logo-github"
                        size="large"
                      />
                    </span>
                  </div>
                  <div
                    class="icon_title"
                  >
                    Github
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="quotes_wrap"
        >
          <div
            class="quotes_mask"
          />
          <div
            class="quotes_content"
          >
            <div
              class="quotes_title"
            >
              Quotes of the Day
            </div>
            <div
              class="quotes_subtitle"
            >
              <a
                href="http://source.unsplash.com"
              >
                Random Background Image from Unsplash
              </a>
            </div>
            <div
              class="quotes_cards"
            >
              <div
                class="ant-row ant-row-center"
                style="margin-left: -12px; margin-right: -12px;"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="ant-layout-footer"
      style="text-align: center; background: rgb(68, 79, 96); padding: 3rem 1.5rem 6rem; color: rgb(255, 255, 255);"
    >
      Made with 
      <i
        class="fa fa-heart pulse"
      />
       by 
      <a
        href="http://harrywang.me"
        rel="noopener noreferrer"
        target="_blank"
      >
        Harry Wang
      </a>
    </footer>
  </section>
</DocumentFragment>
`;
